gp-includes/route.php

```diff
@@ -118,6 +118,56 @@ public function cannot_and_redirect( $action, $object_type = null, $object_id =
 		return false;
 	}
 
+	/**
+	 * Verifies a nonce for a route.
+	 *
+	 * @since 2.0.0
+	 *
+	 * @param string $action Context for the created nonce.
+	 * @return bool False if the nonce is invalid, true if valid.
+	 */
+	public function verify_nonce( $action ) {
+		if ( empty( $_REQUEST['_gp_route_nonce'] ) ) {
+			return false;
+		}
+
+		if ( ! wp_verify_nonce( $_REQUEST['_gp_route_nonce'], $action ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Verifies a nonce for a route and redirects in case the nonce is invalid.
+	 *
+	 * @since 2.0.0
+	 *
+	 * @param string      $action Context for the created nonce.
+	 * @param string|null $url    The URL to redirect. Default: 'null', the referrer.
+	 * @return bool False if the nonce is valid, true if the redirect has happened.
+	 */
+	public function verify_nonce_and_redirect( $action, $url = null ) {
+		if ( $this->verify_nonce( $action ) ) {
+			return false;
```

Comment:
Makes more sense to me, makes it clear the returned false indicates a valid nonce.